### ISSUE TITLE
ERRO[0002] GoError: illegal base64 data at input byte 1496

### ISSUE THREAD

**Issue by senocak** [loadimpact/k6#1509]
## Environment

- k6 version:
- OS and version:
- Docker version and image, if applicable:

## Steps to Reproduce the Problem

1. Getting the "ERRO[0002] GoError: illegal base64 data at input byte 1496" error which is valid token when i use jwt.io to check its valid or not. IT IS VALID
2.
3.

**Comment by MStoykov** [comment]
hi @senocak, 

can you provide us with an input that reproduces it? 
p.s. a k6 version will also be nice although this hasn't been changed since forever

**Comment by MStoykov** [comment]
Closing and marking as invalid as nothing was provided to reproduce or confirm the bug. 
I will reopen it more information is given ;)

**Comment by rkuzmin** [comment]
Hi, here is the problem:
Decode in browser:
```
atob("eyJleHAiOjE3MjY5MTIxODAsInBheWxvYWQiOnsiZGV2aWNlSWQiOiJmNzFkYjhiMTc2MTg4ZTc4ZTkxNzM1NDFjODFkYmI0NSIsImlzQW5vbnltb3VzIjpmYWxzZSwiaXNEZXZlbG9wZXIiOnRydWV9LCJzdWIiOiIwNTkzYWE1MS1hYmJmLTRhYWMtYjExMy05ZTNjMjZmMDYyNDcifQ")
'{"exp":1726912180,"payload":{"deviceId":"f71db8b176188e78e9173541c81dbb45","isAnonymous":false,"isDeveloper":true},"sub":"0593aa51-abbf-4aac-b113-9e3c26f06247"}'
```

Same code in K6
```
const payload = JSON.parse(encoding.b64decode(parts[1]));

ERRO[0004] GoError: illegal base64 data at input byte 212
	at go.k6.io/k6/js/modules/k6/encoding.(*Encoding).b64Decode-fm (native)
	at decodeJWT (/script.js:17:48(29))
	at setup (/script.js:71:24(113))  hint="script exception"
```

**Comment by mstoykov** [comment]
Hi @rkuzmin, thank you for commenting. It probably would've been nicer to be a new issue as this one is 4 years old.

With your provided data, it is reproducible, and if you look at the base64 encoding string there is no padding `=` - at the end. Which usually means that it is standard. I did try with adding `rawstd` as the format and got a different error. Which took me a while to realize meant that b64decode likely doesn't return a string that is directly passable to `JSON.parse` as base64 encode is usually meant to encode binary data as only strings. 

So after that I needed to tell b64decode to return me a string it seems to be working as intended.

```javascript
import encoding from "k6/encoding";
let s = "eyJleHAiOjE3MjY5MTIxODAsInBheWxvYWQiOnsiZGV2aWNlSWQiOiJmNzFkYjhiMTc2MTg4ZTc4ZTkxNzM1NDFjODFkYmI0NSIsImlzQW5vbnltb3VzIjpmYWxzZSwiaXNEZXZlbG9wZXIiOnRydWV9LCJzdWIiOiIwNTkzYWE1MS1hYmJmLTRhYWMtYjExMy05ZTNjMjZmMDYyNDcifQ"

const payload = JSON.parse(encoding.b64decode(s, "rawstd", "s"));

console.log(payload);
export default () => { } // just to not get an error
```

More info can be found in the [documentation](https://grafana.com/docs/k6/latest/javascript-api/k6-encoding/b64decode/) as well as other options.
